addons/medical/functions/fnc_vitalLoop.sqf

```diff
@@ -34,6 +34,24 @@ if (!local _unit) exitWith {
 // Handle unit vitals
 [_unit, CBA_missionTime - _lastTime] call FUNC(handleUnitVitals);
 
+// remove old actions
+[_unit] call FUNC(clearExpired);
+
+// Check damage or bleeing for Splint
+private _damage = _unit getVariable [QGVAR(bodyPartStatus), [0, 0, 0, 0, 0, 0]];
+private _tourniquets = _unit getVariable [QGVAR(tourniquets), [0, 0, 0, 0, 0, 0]];
+private _openWounds = _unit getVariable [QGVAR(openWounds), []];
+private _splint = _unit getVariable [QGVAR(samSplint), [[0, 0], [0, 0], [0, 0], [0, 0], [0, 0], [0, 0]]];
+for "_i" from 0 to 5 do {
+    private _hasDamage = (_damage select _i) > 0;
+    private _hasTourniquet = (_tourniquets select _part) != 0;
+    private _bleedingWound = {((_x select 2) isEqualTo _part) && ((_x select 4) * (_x select 3) > 0)} count ([_openWounds, _bandagedWounds] select _bandagedAlso) > 0;
+
+    if ((_hasDamage || (!_hasTourniquet && _bleedingWound)) && ((_splint select _i select 0) > 0)) then {
```

Comment:
Should use lazy evaluation in this case, for the `((_splint select _i select 0) > 0)` part.

Reply:
Well since the `treatmentSAMSplintDamaged` is executed via `call`command, I thought using lazy evaluation will have almost no point, but rather harm optimization for loading variables unnecessarily.

Reply:
>Well since the `treatmentSAMSplintDamaged` is executed via `call` command,

No idea what that has to do with this.
>but rather harm optimization for loading variables unnecessarily.

Using no lazy evaluation will harm the performance. Not the opposite.
there is absolutely no "optimization" in SQF like in other real programming languages if that's what you are referring to with "optimization"